### PR TITLE
feat: add typography-hint-xs and typography-error-xs classes

### PR DESCRIPTION
**Pull request by Szymon-dziewonski**
# Related issue

https://github.com/vuestorefront/storefront-ui/issues/2981

# Scope of work

Add new types for typography plugin,  `typography-hint-xs` and `typography-error-xs`
Figma: https://www.figma.com/file/CWOkbpne0tDpSenT4ZEUTQ/%F0%9F%9B%A0-SFUI-2-%7C-Design-Kit-v2.6-(development)?node-id=19478%3A53189&mode=dev

Fix wrong styling or missing gaps in hint, replace text-xs into typography-hint-xs
screencapture-localhost-8080-v2-vue-customization-typography-html-2023-10-04-15_45_30

# Screenshots of visual changes

# Checklist

- [x] Self code-reviewed
- [ ] Changes documented
- [ ] Semantic HTML
- [ ] SSR-friendly
- [ ] Caching friendly
- [ ] a11y for WCAG 2.0 AA
- [ ] examples created
- [ ] blocks created
- [ ] cypress tests created

### CODE REVIEW

File: apps/preview/next/pages/showcases/Textarea/TextareaAutoresize.tsx [hunk 1]
```diff
@@ -17,13 +17,9 @@ export default function TextareaAutoresize() {
     <>
       <label>
         <span className="typography-text-sm font-medium">Description</span>
-        <SfTextarea ref={textareaRef} className="w-full h-max-[500px]" size="sm" aria-label="Label size sm" />
+        <SfTextarea ref={textareaRef} className="w-full h-max-[500px] block" size="sm" aria-label="Label size sm" />
```
Comment (FRSgit) [inline apps/preview/next/pages/showcases/Textarea/TextareaAutoresize.tsx:20]: Question: why is that needed?

File: apps/preview/next/pages/showcases/Textarea/TextareaCharacters.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ export default function TextareaWithLimit() {
           invalid={invalid}
           disabled={disabled}
           onChange={onChange}
-          className="w-full mt-0.5"
+          className="w-full mt-0.5 block"
```
Comment (FRSgit) [inline apps/preview/next/pages/showcases/Textarea/TextareaCharacters.tsx:36]: Question: why is that block here?

File: apps/preview/nuxt/pages/showcases/Checkout/CheckoutAddressForm.vue [hunk 3]
```diff
@@ -84,6 +86,7 @@ const streetIsValid = ref(true);
 
 const onSubmit = (e: Event) => {
   /* your submit handler, e.g.: */
+  console.log(e);
```
Comment (FRSgit) [inline apps/preview/nuxt/pages/showcases/Checkout/CheckoutAddressForm.vue:89]: Question: why is that needed?

File: apps/preview/next/pages/showcases/Textarea/TextareaCharacters.tsx (hunk 2 above)
Comment (Szymon-dziewonski) [reply apps/preview/next/pages/showcases/Textarea/TextareaCharacters.tsx:36]: textarea i inline-block by default so there is unexpected gap below textarea therefor wrong ui

File: apps/preview/next/pages/showcases/Textarea/TextareaAutoresize.tsx (hunk 1 above)
Comment (Szymon-dziewonski) [reply apps/preview/next/pages/showcases/Textarea/TextareaAutoresize.tsx:20]: same as next comment

File: apps/preview/nuxt/pages/showcases/Checkout/CheckoutAddressForm.vue (hunk 3 above)
Comment (Szymon-dziewonski) [reply apps/preview/nuxt/pages/showcases/Checkout/CheckoutAddressForm.vue:89]: its not needed, my leftover

**Review by FRSgit — APPROVED** [review]
